pootle/core/utils/version.py

```diff
@@ -0,0 +1,214 @@
+            if part in CANDIDATE_MARKERS][0]
+
+
+def _get_candidate(version):
+    """Returns the candidate. One of alpha, beta, rc or final.
+
+    >>> _get_candidate((0, 1, 2, 'rc', 1))
+    'rc'
+    """
+    return version[_get_candidate_pos(version)]
+
+
+def _get_version_string(parts):
+    """Returns an X.Y.Z version from the list of version parts.
+
+    >>> _get_version_string((1, 1, 0))
+    '1.1.0'
+    >>> _get_version_string((1, 1, 0, 1))
+    '1.1.0.1'
+    """
+    return '.'.join(str(x) for x in parts)
+
+
+def get_main_version(version=None):
+    """Returns main version (X.Y[.Z]) from VERSION.
+
+    >>> get_main_version((1, 2, 3, 'alpha', 1))
+    '1.2.3'
+    """
+    version = get_complete_version(version)
+    candidate_pos = _get_candidate_pos(version)
+    return _get_version_string(version[:candidate_pos])
+
+
+def get_major_minor_version(version=None):
+    """Returns X.Y from VERSION.
+
+    >>> get_major_minor_version((1, 2, 3, 'final', 0))
+    '1.2'
+    """
+    version = get_complete_version(version)
+    return _get_version_string(version[:2])
+
+
+def get_complete_version(version=None):
+    """Returns a tuple of the Pootle version. Or the supplied ``version``
+
+    >>> get_complete_version()
+    (1, 2, 3, 'alpha', 0)
+    """
+    if version is None:
+        from pootle import VERSION as version
+
+    return version
+
+
+def get_docs_version(version=None, positions=2):
+    """Return the version used in documentation.
+
+    >>> get_docs_version((1, 2, 1, 'alpha', 0))
+    'dev'
+    >>> get_docs_version((1, 2, 1, 'rc', 2))
+    '1.2'
+    """
+    version = get_complete_version(version)
+    candidate = _get_candidate(version)
+    candidate_pos = _get_candidate_pos(version)
+    if positions > candidate_pos:
+        positions = candidate_pos
+    if _is_development_candidate(version):
+        return 'dev'
+    return _get_version_string(version[:positions])
+
+
+@lru_cache()
+def _shell_command(command):
+    """Return the first result of a shell ``command``"""
+    repo_dir = os.path.dirname(os.path.abspath(__file__))
+
+    command_subprocess = subprocess.Popen(
+        command, stdout=subprocess.PIPE, stderr=subprocess.PIPE, shell=True,
+        cwd=repo_dir, universal_newlines=True
+    )
+
+    return command_subprocess.communicate()[0]
+
+
+def get_git_changeset():
+    """Returns a numeric identifier of the latest git changeset.
+
+    The result is the UTC timestamp of the changeset in YYYYMMDDHHMMSS format.
+    This value isn't guaranteed to be unique, but collisions are very unlikely,
+    so it's sufficient for generating the development version numbers.
+
+    >>> get_git_changset()
+    20150530132219
+    """
+    timestamp = _shell_command('git log --pretty=format:%ct --quiet -1 HEAD')
+    try:
+        timestamp = datetime.datetime.utcfromtimestamp(int(timestamp))
+    except ValueError:
+        return None
+    return timestamp.strftime('%Y%m%d%H%M%S')
+
+
+def get_git_branch():
```

Comment:
If you want to keep it then change it to return None for no branch. But i don't see the use; a hash should be enough.